python/pip.bzl

```diff
@@ -45,26 +45,46 @@ def pip_install(requirements, name = "pip", **kwargs):
     Args:
       requirements: A 'requirements.txt' pip requirements file.
       name: A unique name for the created external repository (default 'pip').
+      repo_prefix: prefix for the packages generated in the external repository
       **kwargs: Keyword arguments passed directly to the `pip_repository` repository rule.
     """
 
     # Just in case our dependencies weren't already fetched
     pip_install_dependencies()
 
+    if repo_prefix == None:
+        repo_prefix = "pypi__"
+
     pip_repository(
         name = name,
         requirements = requirements,
+        repo_prefix = repo_prefix,
         **kwargs
     )
 
-def pip_parse(requirements_lock, name = "pip_parsed_deps", **kwargs):
+def pip_parse(requirements_lock, name = "pip_parsed_deps", repo_prefix = None, **kwargs):
+    """Generate external repositories for each entry in a requirements lockfile.
+
+    Args:
+      requirements_lock: a requirements lockfile, as generated by e.g. pip-compile
+      name: name for the initial repository that generates
+        repositories for each entry of the lockfile
+      repo_prefix: prefix for the generated external repositories
+      **kwargs: keyword arguments passed directly to the `pip_repository` repository rule.
+
+    """
+
     # Just in case our dependencies weren't already fetched
     pip_install_dependencies()
 
+    if repo_prefix == None:
+        repo_prefix = "{name}_pypi__".format(name = name)
```

Comment:
Without the parent repo name in the repo_prefix these repositories won't be unique, and if users have multiple instances of pip_parse they could collide. I'm not sure I'm in favor of allowing that.

Reply:
We could require a pattern like `<repo-name>_<pkg-name>`-my biggest objective is to have short, memorable names that are easy to refer to by label.

Reply:
i.e. most of all I want to solve https://github.com/bazelbuild/rules_python/issues/414, which just so happens to be easier in the `pip_parse` case than the `pip_install` case...

Reply:
@person142 if we can make these repo names unique using `<repo-name>_<pkg-name>` Where <pkg-name> is the [PEP503 canonical package name](https://www.python.org/dev/peps/pep-0503/#normalized-names) I think thats a good soln.

Reply:
👍 Should have some time to make the change today.